draft-packages/menu/KaizenDraft/Menu/MenuDropdown.tsx

```diff
@@ -60,12 +60,14 @@ const MenuDropdown = ({
       if (
         popperElement &&
         e.target instanceof Node &&
-        !popperElement.contains(e.target)
+        !popperElement.contains(e.target) &&
+        referenceElement !== e.target &&
+        !referenceElement?.contains(e.target)
```

Comment:
Because we’re calling this all the time now, we need to check that the `referenceElement` and its children aren’t being clicked so that we can leave the rest of the event handling/toggling behaviour alone.